diff --git a/Compositor/cmake/Findwayland-client.cmake b/Compositor/cmake/Findwayland-client.cmake
index abc02054..a7ef5abc 100644
--- a/Compositor/cmake/Findwayland-client.cmake
+++ b/Compositor/cmake/Findwayland-client.cmake
@@ -32,7 +32,7 @@ endif()
 
 find_package(PkgConfig)
 
-pkg_check_modules(wayland-client wayland-client ${FIND_MODE} IMPORTED_TARGET)
+pkg_check_modules(wayland-client wayland-client ${FIND_MODE} IMPORTED_TARGET GLOBAL)
 
 if(wayland-client_FOUND)
 add_library(wayland-client::wayland-client ALIAS PkgConfig::wayland-client)